ci/docker/golang/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM debian:buster-slim
+# TODO: Change base image to ubuntu:20.04 on 2024-06-30
+# Then to debian:bullseye-slim on 2025-05-29
+# Then to ubuntu:22.04 on 2026-06-30
+# Then to debian:bookworm-slim on 2027-06-01
+# Then to ubuntu:24:04 on 2028-06-30
+
+LABEL org.opencontainers.image.source=https://github.com/NordSecurity/nordvpn-linux
+
+# Based on: https://github.com/docker-library/golang/blob/a6fd6eceb0cb26da2fceefb4353768c472f84420/1.21/bookworm/Dockerfile
```

Comment:
https://github.com/docker-library/golang/blob/a6fd6eceb0cb26da2fceefb4353768c472f84420/1.20/bullseye/Dockerfile
is probably the correct link.

I also think it would make sense to simply copy everything below the line 1 and specify it instead of modifying the generated code

Reply:
The `github.com/docker-library/golang` Dockerfile installs some dependencies (for example `g++`) which add hundreds of MB to image size, but are not needed for us. The builder image, which is downloaded several times in the pipeline, built from the unmodified `github.com/docker-library/golang` Dockerfile is 500MB larger then built from our `ci/docker/golang/Dockerfile`.